ttnn/cpp/ttnn/operations/reduction/generic/device/kernels/dataflow/reader_unary_transpose_wh_interleaved_input_cols_partitioned.cpp

```diff
@@ -36,23 +37,38 @@ void kernel_main() {
 
     uint32_t w = curr_col_in_batch;
 
-    // this reader will read a NHW tensor in NWH order
-    for (uint32_t i = 0; i < num_cols; i++) {
+    for (uint32_t i = 0; i < num_cols; i += row_chunk) {
+        uint32_t chunk_end = std::min(i + row_chunk, num_cols);
         uint32_t curr_id = col_start_tile_id;
-        for (uint32_t j = 0; j < Ht; j++) {
-            cb_reserve_back(cb_id_in0, onetile);
-            uint32_t l1_write_addr = get_write_ptr(cb_id_in0);
-            noc_async_read_tile(curr_id, s, l1_write_addr);
-            noc_async_read_barrier();
-            cb_push_back(cb_id_in0, onetile);
-            curr_id += Wt;  // stride in H
-        }
-        w++;
-        if (w == Wt) {
-            col_start_tile_id = curr_id - Wt + 1;
-            w = 0;
-        } else {
-            col_start_tile_id++;
+        uint32_t reset_curr_id = curr_id;
+        uint32_t reset_w = w;
+        uint32_t reset_col_start = col_start_tile_id;
+
+        // row wise read for a chunk of columns
```

Comment:
Please add comments about how the reset variables work. This logic is complicated.

Reply:
Explained the reasoning behind reset variables, with comments on each one individually.